zerver/lib/actions.py

```diff
@@ -1876,12 +1878,39 @@ def extract_recipients(s: Union[str, Iterable[str]]) -> List[str]:
     if not isinstance(data, list):
         raise ValueError("Invalid data type for recipients")
 
-    recipients = data
+    if not data:
+        # We don't complain about empty message recipients here
+        return data
 
-    # Strip recipients, and then remove any duplicates and any that
-    # are the empty string after being stripped.
-    recipients = [recipient.strip() for recipient in recipients]
-    return list(set(recipient for recipient in recipients if recipient))
+    if isinstance(data[0], str):
+        if data[0].strip().isdigit():
+            recipients = extract_user_ids(data)  # type: Union[List[str], List[int]]
```

Comment:
Two questions about this.  First, we need a try/except block here, since otherwise `extract_user_ids` can cause us to 500 if the input is mixed.

Second, do we actually need to support string-format user IDs?  I don't see a reason to allow that, in which case we can just delete this block.

Reply:
(There's a good use case for being fast-and-loose with comma-separated emails, for convenience of typing, but with user IDs, you're asking the client to marshall the data into a format for you, so you might as well enforce ti being the right format)